crates/bevy_ecs/src/query/filter.rs

```diff
@@ -938,6 +941,10 @@ impl<T: Component> QueryFilter for Changed<T> {
 ///
 /// [`Added`] and [`Changed`] works with entities, and therefore are not archetypal. As such
 /// they do not implement [`ArchetypeFilter`].
+#[diagnostic::on_unimplemented(
+    message = "`{Self}` is not an archetypical filter",
```

Comment:
```suggestion
    message = "`{Self}` is not a query filter that relies only on the presence/absence of components",
```

Reply:
Maybe "is not an archetypal filter" along with a note explaining that an archetypal filter is a query filter that only looks at an entity's archetype, e.g. With/Without?

To me, that sounds more natural and inline with `std`'s own diagnostics than an error message of "is not a query filter that relies only on the presence/absence of components" does.

Reply:
I'll update the `message` to _"`{Self}` is not a valid `Query` filter based on archetype information"_ and add a `note` _"an `ArchetypeFilter` typically uses a combination of `With<T>` and `Without<T>` statements"_ to better indicate what we're expecting. I'm also adding a similar `note` to the `QueryFilter` trait as well.